htlcswitch/failure_detail.go

```diff
@@ -1,5 +1,13 @@
 package htlcswitch
 
+// FailureDetail is an interface implemented by failures that occur on
+// our incoming or outgoing link, or within the switch itself.
+type FailureDetail interface {
+	// FailureString returns the string representation of a failure
+	// detail.
+	FailureString() string
```

Comment:
Maybe this can be an empty marker interface? If they get a method named `String()`, it will work automatically in text formatting. Or is `FailureDetail` really just an `error` type?

Reply:
If we use a marker interface, these enums will need to be wrapped in a struct which embeds it, which makes the already lengthy switch error declaration even longer. I don't think that's worth it for the 2/3 places we have to call `FailureString` for logging?

I'm against using interfaces `error` because that expands what we can use as a `FailureDetail` beyond what it is intended to mean to being a generic error, when it is specifically and error that has occurred on one of our links, that is paired with a wire error.

Reply:
Discussed offline, leaving as is. We accepted that `FailureString` is a way to enforce a strict type in golang's duck typing system.